paig-server/backend/paig/core/middlewares/usage.py

```diff
@@ -98,6 +102,15 @@ async def shutdown_usage_collector():
     await shutdown_usage_collector()  # called on shutdown
 
 
+def capture_event_on_action(event):
+    posthog_client = get_post_hog_client()
```

Comment:
can  we think in the direction to create posthog_client only once like it is done for exporter

Reply:
Although get_post_hog_client is returning a singleton object so it should be fine to call get_post_hog_client() on every capture_event_on_action but for better readability if  get_post_hog_client()  can only be called once it will look good . 
Also this is just a recommendation if it can work like exporter